api/src/opentrons/protocol_api/execute_v3.py

```
@@ -0,0 +1,180 @@
+import logging
+from typing import Any, Dict
+
+from .contexts import ProtocolContext, InstrumentContext
+from . import labware
+from opentrons.types import Point, Location
+
+MODULE_LOG = logging.getLogger(__name__)
+
+
+def load_pipettes_from_json(
+        ctx: ProtocolContext,
+        protocol: Dict[Any, Any]) -> Dict[str, InstrumentContext]:
+    pipette_data = protocol.get('pipettes', {})
```

Accesses to mandatory keys should be dict indexing (`protocol['pipettes']`) to avoid typing issues and fail fast